### ISSUE TITLE
Auth not working

### ISSUE THREAD

**Issue by beatles1** [roflcoopter/viseron#567]
I've enabled the new auth feature with:
```
webserver:
  auth:
```
I was presented the registration screen which I filled out only to be given an "Error loading user" message. I now get this any time I enter the details on the login screen.
![image](https://user-images.githubusercontent.com/1923058/234413496-2e5820f1-d2e7-4718-b72a-8f9ac92fccb7.png)

If I set the logger to debug I get the following:
```
[2023-04-25 22:51:26] [DEBUG   ] [viseron.helpers.storage] - Writing data to /config/.viseron/auth, message repeated 2 times
[2023-04-25 22:51:26] [DEBUG   ] [viseron.components.webserver.api.handlers] - Invalid auth header
[2023-04-25 22:51:26] [WARNING ] [tornado.access] - 401 GET /api/v1/auth/user/90d2c2ace40c4b13af09318bdd2b9a83 (10.5.1.135) 1.62ms
[2023-04-25 22:51:26] [WARNING ] [viseron.components.webserver.request_handler] - Request with failed authentication from 10.5.1.135 for URL: /api/v1/auth/user/90d2c2ace40c4b13af09318bdd2b9a83 Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/112.0
```

If I input an incorrect password I get a relevant message saying the password is wrong. If I disable auth in the config it all loads fine again.

**Comment by roflcoopter** [comment]
Thanks for your report!

Have you blocked cookies by any chance? What happens if you try in incognito mode? Seems something has gotten out of sync

**Comment by beatles1** [comment]
Incognito mode had the same error and cookies should be enabled. Only tried in Firefox so I can test another fresh browser as well when I get a chance.

**Comment by roflcoopter** [comment]
Yeah that would be great! I have only tested in Chrome/Safari/Edge

**Comment by beatles1** [comment]
Unfortunately no luck. I've tried using Chrome to re-register (after removing the .viseron folder which I assume is the right way to reset this) several times. I've tried various passwords and usernames, not sure what else to test.

**Comment by roflcoopter** [comment]
I just did the same (removed the `.viseron` folder) and then restarted Viseron.
I am able to log in correctly, do you have any more logs that show anything?

If you check the browsers Dev tools > Application > Cookies after a login attempt, do you have these cookies?
![image](https://user-images.githubusercontent.com/26493864/234760391-e7d27dea-c3ac-4966-87aa-954b5d5bedca.png)


And in the Local Storage tab, do you have an item called `tokens`?
![image](https://user-images.githubusercontent.com/26493864/234760604-a87e2253-e9d7-4a38-92d7-4049d9475362.png)

**Comment by beatles1** [comment]
Aah, so I've managed to have another look today and all those cookies and storage are there but I think I've worked out why this is happening.

I've got it behind Nginx Proxy Manager as a reverse proxy which has a feature allowing me to add basic HTTP auth to a service. I'd added HTTP auth externally and got it to bypass that for any internal IPs. For whatever reason if I change that to a separate policy allowing only internal IPs and blocking external then the Viseron built in auth works fine. I assume there's some sort of conflict between the basic HTTP auth and Viseron auth, possibly that a header was being read by Nginx and not passed onto the application.

I don't think this is actually an issue, I'd not expect anyone in practice to need HTTP basic auth in front of the built in authentication.
